### PR TITLE
gh-91731: Fix typo in pymacro.h

### PR DESCRIPTION
**Pull request by pablogsal**

### CODE REVIEW

**Comment by vstinner** [comment]
You should use ``gh-91731: `` rather than skipping the issue. You can add it when you merge the PR.

It seems like macOS also has build error with static_assert() :-( https://github.com/python/cpython/issues/91731#issuecomment-1122212919

File: Include/pymacro.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #  define static_assert _Static_assert
 #endif
 
-// static_assert is defined in GLIB from version 2.16. Before it requires
+// static_assert is defined in GLIBC from version 2.16. Before it requires
```
Comment (vstinner) [inline Include/pymacro.h]:
```suggestion
// static_assert is defined in glibc from version 2.16. Before it requires
```

**Review by vstinner — COMMENTED** [review]
Each time I saw the project name mentioned, it was written lower case.

**Comment by pablogsal** [comment]
> ou should use gh-91731: rather than skipping the issue. You can add it when you merge the 

Done :)

**Comment by pablogsal** [comment]
> It seems like macOS also has build error with static_assert() :-( #91731 (comment)

The never ending story :(

**Review by vstinner — APPROVED** [review]
Peeeerfect.

**Pull request merged**